a3p-integration/proposals/z:acceptance/vaults.test.js

```diff
@@ -0,0 +1,201 @@
+  const { body } = JSON.parse(values[0]);
+  const cleanedBody = JSON.parse(body.slice(1));
+  return cleanedBody;
+};
+
+test.serial('attempt to open vaults under the minimum amount', async t => {
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+
+  const mint = '3.0';
+  const collateral = '5.0';
+  await t.throwsAsync(() => openVault(USER1ADDR, mint, collateral));
+
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    activeVaultsAfter.length === activeVaultsBefore.length,
+    'The number of active vaults should remain the same.',
+  );
+});
+
+test.serial('open new vault', async t => {
+  const istBalanceBefore = await getISTBalance(USER1ADDR);
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+
+  const mint = '5.0';
+  const collateral = '10.0';
+  await openVault(USER1ADDR, mint, collateral);
+  await waitForBlock();
+
+  const istBalanceAfter = await getISTBalance(USER1ADDR);
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    istBalanceBefore + 5 === istBalanceAfter,
+    'The IST balance should increase by the minted amount',
+  );
+  t.true(
+    activeVaultsAfter.length > activeVaultsBefore.length,
+    `The number of active vaults should increase after opening a new vault.`,
+  );
+});
+
+test.serial('remove collateral', async t => {
+  const activeVaults = await agops.vaults('list', '--from', USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultDataString = await queryVstorage(vaultPath);
+  let vaultData = getFormattedVaultData(vaultDataString);
+  const collateralBefore = vaultData.locked.value;
+
+  await adjustVault(USER1ADDR, vaultID, { wantCollateral: 1.0 });
+  await waitForBlock();
+
+  vaultDataString = await queryVstorage(vaultPath);
+  vaultData = getFormattedVaultData(vaultDataString);
+  const collateralAfter = vaultData.locked.value;
+
+  t.true(
+    Number(collateralBefore) > Number(collateralAfter),
+    'The vault Collateral should decrease after removing some ATOM',
+  );
+});
+
+test.serial('remove IST', async t => {
+  const activeVaults = await agops.vaults('list', '--from', USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultDataString = await queryVstorage(vaultPath);
+  let vaultData = getFormattedVaultData(vaultDataString);
+  const debtBefore = vaultData.debtSnapshot.debt.value;
+
+  await adjustVault(USER1ADDR, vaultID, { wantMinted: 1.0 });
+  await waitForBlock();
+
+  vaultDataString = await queryVstorage(vaultPath);
+  vaultData = getFormattedVaultData(vaultDataString);
+  const debtAfter = vaultData.debtSnapshot.debt.value;
+
+  t.true(
+    Number(debtAfter) > Number(debtBefore),
```

Comment:
since this is converting to numbers anyway, it can assert that debtAfter is 1 more than debtBefore.
And the same on line 80 above.

Reply:
Thank you for bringing this to my attention.
I replaced `t.true` with `t.is` and made the difference in values explicit in the assertion, which is now also printed to the console.
Change addressed on commit [1894140](https://github.com/Agoric/agoric-sdk/pull/10123/commits/1894140c246d6935f41805d4da3b22cf592abdc2)